libgadget/domain.c

```diff
@@ -78,31 +73,29 @@ struct local_particle_data
     int64_t Cost;
 };
 
-struct task_data * Tasks;
-
 static int
 order_by_key(const void *a, const void *b);
 static void
 mp_order_by_key(const void * data, void * radix, void * arg);
 
 static void
-domain_assign_balanced(int64_t * cost);
+domain_assign_balanced(Domain * domain, int64_t * cost);
```

Comment:
The word Domain is confusing here. It can refer to the domain module or a single domain.

Here are three suggestions:

`DomainMaster * dmaster, DomainWizard * dwizard, DomainController * dcontroller`

Reply:
That is true. DomainWizard sounds too Harry Potter to me. How about DomainDecomp * ddecomp?
Otherwise I like DomainController best.